graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ConfigGenerator.java

```diff
@@ -0,0 +1,304 @@
+
+/**
+ * This class is used to generate connection config related types in the ballerina config types file.
+ */
+public class ConfigGenerator {
+    private static ConfigGenerator configGenerator = null;
+    private static final String CONNECTION_CONFIG = "ConnectionConfig";
+    private static final String CLIENT_HTTP1_SETTINGS = "ClientHttp1Settings";
+    private static final String PROXY_CONFIG = "ProxyConfig";
+    private static final String KEEP_ALIVE = "KeepAlive";
+    private static final String CHUNKING = "Chunking";
+    private static final String KEEPALIVE_AUTO = "KEEPALIVE_AUTO";
+    private static final String KEEPALIVE_ALWAYS = "KEEPALIVE_ALWAYS";
+    private static final String KEEPALIVE_NEVER = "KEEPALIVE_NEVER";
+    private static final String CHUNKING_AUTO = "CHUNKING_AUTO";
+    private static final String CHUNKING_ALWAYS = "CHUNKING_ALWAYS";
+    private static final String CHUNKING_NEVER = "CHUNKING_NEVER";
+    private static final String CLIENT_HTTP2_SETTINGS = "ClientHttp2Settings";
+    private static final String CACHE_CONFIG = "CacheConfig";
+    private static final String CACHING_POLICY = "CachingPolicy";
+    private static final String CACHE_CONTROL_AND_VALIDATORS = "CACHE_CONTROL_AND_VALIDATORS";
+    private static final String RFC_7234 = "RFC_7234";
+    private static final String COMPRESSION = "Compression";
+    private static final String COMPRESSION_AUTO = "COMPRESSION_AUTO";
+    private static final String COMPRESSION_ALWAYS = "COMPRESSION_ALWAYS";
+    private static final String COMPRESSION_NEVER = "COMPRESSION_NEVER";
+    private static final String CIRCUIT_BREAKER_CONFIG = "CircuitBreakerConfig";
+    private static final String ROLLING_WINDOW = "RollingWindow";
+    private static final String RETRY_CONFIG = "RetryConfig";
+    private static final String RESPONSE_LIMIT_CONFIGS = "ResponseLimitConfigs";
+    private static final String CLIENT_SECURE_SOCKET = "ClientSecureSocket";
+    private static final String TRUST_STORE = "TrustStore";
+    private static final String KEY_STORE = "KeyStore";
+    private static final String CERT_KEY = "CertKey";
+    private static final String CERT_VALIDATION_TYPE = "CertValidationType";
+    private static final String PROTOCOL = "Protocol";
+    private static final String POOL_CONFIGURATION = "PoolConfiguration";
+    private static final String HTTP_VERSION = "HttpVersion";
+    private static final String BEARER_TOKEN_CONFIG = "BearerTokenConfig";
+    private static final String CREDENTIALS_CONFIG = "CredentialsConfig";
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigGenerator.class);
+
+    public static ConfigGenerator getInstance() {
+        if (configGenerator == null) {
+            configGenerator = new ConfigGenerator();
+        }
+        return configGenerator;
+    }
+
+    public SyntaxTree generateConfigRecord(AuthConfig authConfig) throws IOException {
```

Comment:
Let's make this function private if not used anywhere else and expose only the single function `generateSrc` as public.

Reply:
Updated in https://github.com/ballerina-platform/graphql-tools/commit/127efce1497eae87b00a2136832a9afc7dd0e3ef